sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -852,6 +855,9 @@ COMMIT: 'COMMIT';
 ROLLBACK: 'ROLLBACK';
 MACRO: 'MACRO';
 IGNORE: 'IGNORE';
+BOTH: 'BOTH';
+LEADING: 'LEADING';
+TRAILING: 'TRAILING';
```

Comment:
Add these to `TableIdentifierParserSuite`?

Reply:
Sure, I will add these three keywords in the hiveNonReservedKeyword array. The test case "table identifier - non reserved keywords" will cover the testing, so I will not add new test case here.